arbnode/sequencer.go

```diff
@@ -206,14 +283,22 @@ func (s *Sequencer) PublishTransaction(ctx context.Context, tx *types.Transactio
 	}
 }
 
-func (s *Sequencer) preTxFilter(_ *params.ChainConfig, _ *types.Header, statedb *state.StateDB, _ *arbosState.ArbosState, _ *types.Transaction) error {
+func (s *Sequencer) preTxFilter(_ *params.ChainConfig, header *types.Header, statedb *state.StateDB, _ *arbosState.ArbosState, tx *types.Transaction, sender common.Address) error {
+	if s.nonceCache.GetSize() > 0 {
+		stateNonce := s.nonceCache.Get(header, statedb, sender)
+		err := MakeNonceError(sender, tx.Nonce(), stateNonce)
+		if err != nil {
+			return err
```

Comment:
not necessarily for this PR: metrics?

Reply:
I've gone ahead and added some. Let me know if you think of any others that'd be useful.

Reply:
I'm thinking metrics for BeginNewBlock and Finalize.. comparing these to cacheCleared will help us detect problems there.